File: scripts/parame.py
# -*- coding: utf-8 -*-
import pandas as pd
import cepy as cp

# Leitura do arquivo de dados
matriz = pd.read_csv("especies.csv", header=0, index_col=0, encoding="utf8")
names = cp.make_cepnames(matriz.index)
names = [name[0:4] + ' ' + name[4:8] for name in names]
matriz.index = names

# Chamada da função definida pelo usuário parameters()
from params import parameters
p = parameters(matriz, freq=True, dom=True, tofile=False)

# Número total de indivíduos
t = p['abundance'].sum()

# Número total de espécies
s = len(matriz)

# Total de espécies por categoria
a = len(p[p['category'] == "Acessoria"])
d = len(p[p['category'] == "Acidental"])
c = len(p[p['category'] == "Constante"])

# Saída dos resultados
from tabulate import tabulate
print(tabulate(p, headers='keys', tablefmt='psql', showindex=False, 
               floatfmt=("%s", "%d", ".2f", "%s", "%d", ".2f")))

print("Total de indivíduos = " + str(t))
print("Total de espécies   = " + str(s))
print("Espécies constantes = " + str(c))
print("Espécies acessórias = " + str(a))
print("Espécies acidentais = " + str(d))

File: scripts/r/parame.R
# Carregamento da funcao "parameters"
source("params.R")

# Numero de digitos no formato de saida
options(digits=4, width=90)

# Leitura do arquivo de dados
x <- read.csv("especies.csv", header=TRUE, as.is=TRUE)
#x <- read.table("especies.csv", header=TRUE, sep=",", na.strings="NA", dec=".", strip.white=TRUE, as.is=TRUE)

# Chamada da funcao "parameters"
p <- parameters(x, freq=TRUE, dom=TRUE, tofile=FALSE)

# Numero total de individuos
i <- sum(p$abundance)

# Numero total de especies
s <- nrow(x)

# Total de especies por categoria
a <- length(which(p$category == "Acessoria"))
d <- length(which(p$category == "Acidental"))
c <- length(which(p$category == "Constante"))

# Saida dos resultados
linha <- paste("+", strrep("-", 81), "+", sep="")
cat(linha, fill=TRUE)
cat(knitr::kable(p), fill=TRUE)
cat(linha, fill=TRUE)

cat(paste("Total de individuos = ", i, sep=""), fill=TRUE)
cat(paste("Total de especies   = ", s, sep=""), fill=TRUE)
cat(paste("Especies constantes = ", c, sep=""), fill=TRUE)
cat(paste("Especies acessorias = ", a, sep=""), fill=TRUE)
cat(paste("Especies acidentais = ", d, sep=""), fill=TRUE)